### PR TITLE
MRG: Add outline='skirt' option to plot_topomap

### PR DESCRIPTION
**Pull request by wmvanvliet**
In some dense EEG montages, electrodes are at the side / underside of the
head. When making a topomap plot, it can help interpretation to place
these electrodes (and corresponding interpolated data) outside the head
circle. Setting outline='skirt' allows you to do this.

figure_1

To see an example of the API:

https://gist.github.com/wmvanvliet/6d7c78ea329d4e9e1217

### CODE REVIEW

File: mne/viz/topomap.py [hunk 1]
```diff
@@ -242,12 +244,12 @@ def plot_projs_topomap(projs, layout=None, cmap='RdBu_r', sensors=True,
 
 
 def _check_outlines(pos, outlines, head_pos=None):
-    """Check or create outlines for topoplot
+    """Check or create outlines for 1opoplot
```
Comment (Eric89GXL) [inline mne/viz/topomap.py]: whoops

**Comment by agramfort** [comment]
Can you add a test ?

**Comment by agramfort** [comment]
Nothing complex just for coverage

**Comment by agramfort** [comment]
set the PR to MRG when reading. thx

File: mne/decoding/base.py [hunk 2]
```diff
@@ -415,14 +415,15 @@ def plot_patterns(self, info, times=None, ch_type=None, layout=None,
...
-            If dict, the 'autoshrink' (bool) field will trigger automated
-            shrinking of the positions due to points outside the outline.
-            Moreover, a matplotlib patch object can be passed for
+        outlines : 'head' | 'skirt' | dict | None
+            The outlines to be drawn. If 'head', the default head scheme will
+            be drawn. If 'skirt' the head scheme will be drawn, but sensors are
+            allowed to be plotted outsize of the head circle. If dict, each key
+            refers to a tuple of x and y positions. The values in 'mask_pos'
+            will serve as image mask. If None, nothing will be drawn.  Defaults
+            to 'head'. If dict, the 'autoshrink' (bool) field will trigger
```
Comment (jona-sassenhagen) [inline mne/decoding/base.py]: You have two "If dict" lines, wouldn't it be a bit more readable to condense them into one? E.g.,

```
The outlines to be drawn. If 'head', the default head scheme will
be drawn. If 'skirt', the head scheme will be drawn, but sensors are
allowed to be plotted outsize of the head circle. If None, nothing 
will be drawn. If dict, each key refers to a tuple of x and y positions
and the 'autoshrink' (bool) field will trigger automated shrinking of
the positions due to points outside the outline. The values in 
'mask_pos' will serve as image mask. Moreover, a matplotlib patch 
object can be passed for advanced masking options, either directly 
or as a function that returns patches (required for multi-axis plots).
 Defaults to 'head'. 
```

File: mne/viz/topomap.py [hunk 3]
```diff
@@ -273,23 +275,42 @@ def _check_outlines(pos, outlines, head_pos=None):
...
+                head_pos['scale'] = 1.0 / (pos.max(axis=0) - pos.min(axis=0))
+            pos *= head_pos['scale']
+
+            # Make the figure encompass slightly more than all points
+            mask_scale = 1.25 * (pos.max(axis=0) - pos.min(axis=0))
+
+            outlines_dict['autoshrink'] = False
+            outlines_dict['mask_pos'] = (mask_scale[0] * head_x,
+                                         mask_scale[1] * head_y)
+            outlines_dict['clip_radius'] = (mask_scale / 2)
```
Comment (jona-sassenhagen) [inline mne/viz/topomap.py]: Not ```2.```?

Comment (wmvanvliet) [inline mne/viz/topomap.py]: mask_scale is guaranteed to be of type float (line 295), but I'll set it to 2. for readability.

**Comment by wmvanvliet** [comment]
@agramfort Have at it!

**Comment by agramfort** [comment]
great !

@wmvanvliet I'll let you push a what's new entry directly in master.

**Comment by ImmanuelSamuel** [comment]
Can you also add outlnes='skirt' for mne.viz.plot_topomap?

**Comment by agramfort** [comment]
please give it a try.

**Pull request merged**